In [14]:
import sys
sys.path.append("..")
from pathlib import Path
import plotly.graph_objects as px
from utils.transcription_utils import get_transcription_info,get_stats,get_total_stats
import csv
import pandas as pd
import time
import tarfile
from tqdm import tqdm

In [15]:
save_path=Path("/media/dataset-harddisk/munikumar/utils/audio_utils/data/kathbath/hard")

In [16]:
dataset_dir=Path("/media/dataset-harddisk/munikumar/hindi_dataset/kathbath_hard/kathbath_hard_raw")

In [17]:
extract_path=Path("/media/dataset-harddisk/munikumar/hindi_dataset/kathbath_hard/kathbath_hard_raw/extract")

In [18]:
def extract_files(extraction_path):
    for tar_file in dataset_dir.glob("*.tar"):
        start_time = time.time()
        with tarfile.open(tar_file) as tar:
            with tqdm(desc=f"Extracting {tar_file.name} ") as progress_bar:
                member = tar.next()
                while member is not None:
                    progress_bar.update(1)
                    if member.name.startswith("kb_data_noisy_m4a/hindi/"):
                        tar.extract(member, path=str(extraction_path))
                    member = tar.next()
        end_time = time.time()
        execution_time = end_time - start_time
        print(f"{tar_file.name} extraction time: {execution_time:.4f} seconds")

In [19]:
extract_files(extract_path)

Extracting transcripts_n2w.tar : 36it [00:00, 1996.50it/s]


transcripts_n2w.tar extraction time: 0.0199 seconds


Extracting testunk_audio.tar : 11730it [00:00, 14115.07it/s]


testunk_audio.tar extraction time: 0.8333 seconds


Extracting testkn_audio.tar : 29182it [00:02, 12014.32it/s]

testkn_audio.tar extraction time: 2.4342 seconds


In [20]:
def load_transcription(transcription_path):
    transcriptions={}
    with open(transcription_path, "r", encoding="utf-8") as fp:
        for transcription_data in fp.read().splitlines():
            audio_file = transcription_path.parent.name+"/audio/"+transcription_data.split("\t")[0]
            transcription = transcription_data.split("\t")[-1]
            transcriptions[audio_file] = transcription
    return transcriptions

In [21]:
dataset_path=extract_path/"kb_data_noisy_m4a"/"hindi"

In [22]:
test_transcription_data=load_transcription(dataset_path/"test"/"transcription_n2w.txt")
testk_transcription_data=load_transcription(dataset_path/"test_known"/"transcription_n2w.txt")
test_transcription_data.update(testk_transcription_data)
get_transcription_info(test_transcription_data,dataset_path,save_path,"test")
get_stats(save_path/"test_non_hindi_transcripts.json")

getting audio info about : hard - test


  0%|          | 3/3850 [00:00<02:59, 21.48it/s]

100%|██████████| 3850/3850 [02:58<00:00, 21.58it/s]


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 6:22:50
Loss duration : 0:0:17
Remaining duration : 6:22:33
Drop by percentage: 0.07

Total audio files : 3850
Files that would be drop: 3
Remaining audio files : 3847
Drop by percentage: 0.08
**************************************************
Non hindi tokes : ['ॊ', '॓', 'ॢ', '॰']


### Test Dataset
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 8:1:8  
Loss duration : 0:0:41  
Remaining duration : 8:0:26  
Drop by percentage: 0.14  

Total audio files : 5059  
Files that would be drop: 6  
Remaining audio files : 5053  
Drop by percentage: 0.12  
**************************************************
Non hindi tokes : ['ऍ', 'ऽ', '॓']

In [10]:
valid_transcription_data=load_transcription(dataset_path/"valid"/"transcription_n2w.txt")
get_transcription_info(valid_transcription_data,dataset_path,save_path,"dev")
get_stats(save_path/"dev_non_hindi_transcripts.json")

getting audio info about : normalize - dev


  0%|          | 0/3229 [00:00<?, ?it/s]

100%|██████████| 3229/3229 [03:20<00:00, 16.07it/s]


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 5:0:33
Loss duration : 0:0:14
Remaining duration : 5:0:18
Drop by percentage: 0.08

Total audio files : 3229
Files that would be drop: 2
Remaining audio files : 3227
Drop by percentage: 0.06
**************************************************
Non hindi tokes : ['ऍ', 'ॆ']


### Dev Dataset
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 5:0:33  
Loss duration : 0:0:14  
Remaining duration : 5:0:18  
Drop by percentage: 0.08  

Total audio files : 3229  
Files that would be drop: 2  
Remaining audio files : 3227  
Drop by percentage: 0.06  
**************************************************
Non hindi tokes : ['ऍ', 'ॆ']

In [11]:
train_transcription_data=load_transcription(dataset_path/"train"/"transcription_n2w.txt")
get_transcription_info(train_transcription_data,dataset_path,save_path,"train")
get_stats(save_path/"train_non_hindi_transcripts.json")

getting audio info about : normalize - train


100%|██████████| 93024/93024 [2:06:06<00:00, 12.29it/s]  


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 137:19:4
Loss duration : 0:5:10
Remaining duration : 137:13:54
Drop by percentage: 0.06

Total audio files : 93024
Files that would be drop: 46
Remaining audio files : 92978
Drop by percentage: 0.05
**************************************************
Non hindi tokes : ['ऌ', 'ऍ', 'ऎ', 'ऺ', 'ॆ', 'ॊ', 'ॐ', '॓', 'ॠ', 'ॡ', 'ॢ', '॰']


### Train Dataset
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 137:19:4  
Loss duration : 0:5:10  
Remaining duration : 137:13:54  
Drop by percentage: 0.06  

Total audio files : 93024  
Files that would be drop: 46  
Remaining audio files : 92978  
Drop by percentage: 0.05  
**************************************************
Non hindi tokes : ['ऌ', 'ऍ', 'ऎ', 'ऺ', 'ॆ', 'ॊ', 'ॐ', '॓', 'ॠ', 'ॡ', 'ॢ', '॰']

In [12]:
get_total_stats(save_path)

**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 150:20:45
Loss duration : 0:6:6
Remaining duration : 150:14:39
Drop by percentage: 0.07

Total audio files : 101312
Files that would be drop: 54
Remaining audio files : 101258
Drop by percentage: 0.05


### Total Dataset
**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 150:20:45  
Loss duration : 0:6:6  
Remaining duration : 150:14:39  
Drop by percentage: 0.07  

Total audio files : 101312  
Files that would be drop: 54  
Remaining audio files : 101258  
Drop by percentage: 0.05  

## Kathbath Clean
test: 1929  
train: 93024  
valid: 3229  
test_known: 3130  
total: 1_01_312  

## Kathbath Hard
test: 816  
test_known: 3034  
total: 3850